### ISSUE TITLE
Aggregate Unknown type: class de.bwaldvogel.mongo.backend.Missing

### ISSUE THREAD

**Issue by troydm** [bwaldvogel/mongo-java-server#148]
When running aggregate using Spring Mongo such as:

```
Aggregation agg = newAggregation(match(new Criteria("name").exists(true).ne(null)),
                                                       group("name").last("updatedBy").as("updatedBy"));
mongoTemplate.aggregate(agg, User.class, User.class).getMappedResults();
```

I get 
```
java.lang.IllegalArgumentException: Unknown type: class de.bwaldvogel.mongo.backend.Missing
	at de.bwaldvogel.mongo.wire.bson.BsonEncoder.determineType(BsonEncoder.java:201) ~[mongo-java-server-core-1.31.0.jar:na]
```

Note that Documents don't contain any updatedBy fields in any of the rows

This issue is probably somehow related to https://github.com/bwaldvogel/mongo-java-server/issues/111 however as I'm using latest 1.31.0 version it seems it is not fixed

**Comment by bwaldvogel** [comment]
Thanks for reporting. This should be fixed in acd8ee9d98aa47fa1bbde37408b64017a4e23065.

**Comment by bwaldvogel** [comment]
The fix is available in version `1.32.0`

**Comment by troydm** [comment]
@bwaldvogel thx, upgrading to 1.32.0 fixed the problem
